spec/support/solidus_stripe/webhook/event_with_context_factory.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "solidus_stripe/webhook/event"
+require "stripe"
+
+module SolidusStripe
+  module Webhook
+    # Factory to create a webhook event along with its context.
+    #
+    # It allows to create Stripe webhook from different sources (hash, stripe
+    # object) in different representations (json, stripe event object, header).
+    #
+    # The context for a event is composed by the timestamp and its secret, which
+    # in turn affect the header representation.
+    class EventWithContextFactory
+      def self.from_hash(hash:, timestamp: Time.zone.now, secret: "whsec_123")
+        new(hash: hash, timestamp: timestamp, secret: secret)
+      end
+
+      def self.from_object(object:, type:, timestamp: Time.zone.now, secret: "whsec_123")
+        hash_base = hash_base(object, type)
+        hash = hash_base.merge("webhook" => hash_base)
+        new(hash: hash, timestamp: timestamp, secret: secret)
+      end
+
+      def self.hash_base(object, type)
+        {
+          "id" => "evt_3MRUo1JvEPu9yc7w091rP2XV",
+          "object" => "event",
+          "api_version" => "2022-11-15",
+          "created" => 1_674_022_050,
+          "data" => {
+            "object" => object.as_json
+          },
+          "livemode" => false,
+          "pending_webhooks" => 0,
+          "request" => {
+            "id" => "req_3MRUo1JvEPu9yc7w0",
+            "idempotency_key" => "idempotency_key"
+          },
+          "type" => type
+        }
+      end
+      private_class_method :hash_base
+
+      attr_reader :hash, :timestamp, :secret
```

Comment:
nit: Not that instances of this class will ever be used as hash-keys, but `#hash` is a special method name which use might be puzzling at first sight, thoughts on renaming it to `data`, `payload`, or something along those lines?

Reply:
That's an excellent point, and I overlooked that. I renamed it to `#data`, as `#payload` is expected to be the JSON string from the Stripe API (which could be confusing).